lib/ansible/modules/cloud/digital_ocean/digital_ocean_snapshot_facts.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: digital_ocean_snapshot_facts
+short_description: Gather facts about DigitalOcean Snapshot
+description:
+    - This module can be used to gather facts about snapshot facts based upon provided values such as droplet, volume and snapshot id.
+author: "Abhijeet Kasurde (@akasurde)"
+version_added: "2.5"
+options:
+  snapshot_type:
+    description:
+     - Specifies the type of snapshot facts to be retrived.
+     - If set to C(droplet), then facts are gathered related to snapshots based on Droplets only.
+     - If set to C(volume), then facts are gathered related to snapshots based on volumes only.
+     - If set to C(by_id), then facts are gathered related to snapshots based on snapshot id only.
+     - If not set to any of the above, then facts are gathered related to all snapshots.
+    default: 'all'
+    choices: [ 'all', 'droplet', 'volume', 'by_id']
+    required: false
+  snapshot_id:
+    description:
+     - To retrieve information about a snapshot, please specify this as a snapshot id.
+     - If set to actual snapshot id, then facts are gathered related to that particular snapshot only.
+     - This is required parameter, if C(snapshot_type) is set to C(by_id).
+    required: false
+  oauth_token:
```

Comment:
A few days ago **utils/module_docs_fragments/digital_ocean.py** (Ansible Digital Ocean utils) was updated in PR #35934 which introduced generic configurations and documentation that should be used, particularly here this documentation fragment **extends_documentation_fragment: digital_ocean.documentation**(https://github.com/Akasurde/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/digital_ocean.py)

Reply:
Done.